tests/Unit/Middleware/TypedPropertiesTest.php

```diff
@@ -32,9 +35,12 @@ public function testTypedPropertyIsCorrectlyDiscoveredWithPhp74(): void
         $middleware->handle(new \stdClass(), new ObjectWrapper($object), $propertyMap, $jsonMapper);
 
         self::assertTrue($propertyMap->hasProperty('name'));
-        self::assertEquals('string', $propertyMap->getProperty('name')->getType());
-        self::assertEquals(Visibility::PUBLIC(), $propertyMap->getProperty('name')->getVisibility());
-        self::assertFalse($propertyMap->getProperty('name')->isNullable());
+        self::assertThatProperty($propertyMap->getProperty('name'))
+            ->hasType('string')
+            ->hasVisibility(Visibility::PUBLIC())
+            ->isNotNullable()
+            ->isNotArray();
+        self::assertTrue($propertyMap->hasProperty('friends'));
```

Comment:
I think it is smart if we add this assertion we validate the type, visibility, nullable and isArray on it.

Reply:
I would like to, but what are proper values here?

Reply:
You can add:
```
self::assertThatProperty($propertyMap->getProperty('friends'))
            ->hasType('mixed')
            ->hasVisibility(Visibility::PUBLIC())
            ->isNullable()
            ->isArray();
```

The `IsNullable` maybe a bit out of whack as it cannot be null but this is another issue with the TypedProperties middleware which I would rather pickup in a separate PR.

Reply:
fixed